GVFS/GVFS/CommandLine/GVFSVerb.cs

```diff
@@ -887,19 +889,23 @@ protected ReturnCode ExecuteGVFSVerb<TVerb>(ITracer tracer, Action<TVerb> config
                             this.EnlistmentRootPathParameter,
                             verb =>
                             {
-                                verb.Output = writer;
+                                verb.Output = outputWriter ?? writer;
                                 configureVerb?.Invoke(verb);
                             });
                     }
 
-                    tracer.RelatedEvent(
-                        EventLevel.Informational,
-                        typeof(TVerb).Name,
-                        new EventMetadata
-                        {
-                        { "Output", commandOutput.ToString() },
-                        { "ReturnCode", returnCode }
-                        });
+                    EventMetadata metadata = new EventMetadata();
+                    if (outputWriter == null)
+                    {
+                        metadata.Add("Output", commandOutput.ToString());
+                    }
+                    else
+                    {
+                        metadata.Add("Output", $"Check {new TVerb().VerbName} logs for output");

```

Comment:
I don't quite follow this check.  

If `outputWriter` is non-null (e.g. it has been set to `Console.Out`), how is that related to the verb log containing the output?

Reply:
The case I am trying to handle is the `sparse` command running the `DehyrateVerb` which the output is being redirected to `Console.Out` but it creates it's own log file for logging information so if you only look in the `SparseVerb` log you may not see everything that happened during the dehydrate.  That is why it is adding this to the sparse log so you know to look at the log file of the `new TVerb().VerbName` for details.

sparse log will have something like
```
sparse {"Message":"No folders to update in sparse set."}
sparse {"Message":"Found 1 folders to prune."}
DehydrateVerb {"Output":"Check dehydrate logs for output","ReturnCode":7}
Error {"ErrorMessage":"Failed to prune. Exit Code: DehydrateFolderFailures"}
```

Dehydrate log will have the full logs from the verb including
```
Error {"Description":"Deleting '<folder path>'","Error":"The process cannot access the file '<folder path>' because it is being used by another process.","ErrorMessage":"TryIO: Caught exception performing action"}
```

Reply:
Making the entire `TVerb` output logs go to the calling verb's log file is more work since the `ITracer` would need to be passed around and that seemed out of scope for these changes.

Reply:
> but it creates it's own log file for logging information

To confirm, in this example will `sparse`'s log file include the contents of whatever was written to `outputWriter`?  Or is this never recorded by a tracer?

This might be a dumb question but I can't figure out where this:

`{ "Output", commandOutput.ToString() },`

Was ever getting logged (e.g. I don't see it in the `clone` verb when it calls `mount`).

> you may not see everything

Maybe add a comment here, something like:

```
else
{
    // If a parent verb is redirecting the output of its child, include a reminder
    // that the child verb's activity was recorded in its own log file
    metadata.Add("Output", $"Check {new TVerb().VerbName} logs for output");
```

Reply:
>To confirm, in this example will sparse's log file include the contents of whatever was written to outputWriter? Or is this never recorded by a tracer?

No, writing to the `outputWriter` doesn't have anything to do with the `ITracer` that is being used for tracing.  That is what makes this kind of messy because we are redirecting where output is going (to the `Console.Out` of the sparse verb) but the `DehyrateVerb` is still creating its own tracer that is used and writes to the dehydrate log.

>Was ever getting logged (e.g. I don't see it in the clone verb when it calls mount).

Looks like in the case of the `CloneVerb` its tracer is being disposed before the call to `this.Execute<MountVerb>(`.
See https://github.com/microsoft/VFSForGit/blob/c5b4262e38870eb4c648a9bac48ca3c062aad5bc/GVFS/GVFS/CommandLine/CloneVerb.cs#L123-L204

Reply:
> Looks like in the case of the CloneVerb its tracer is being disposed before the call to this.Execute<MountVerb>

Ahh that does sound familiar there was one point in time where we could not have two ETW tracer instances at the same time.  I don't think we still have that restriction but this code was never updated.

> That is what makes this kind of messy because we are redirecting where output is going (to the Console.Out of the sparse verb) but the DehyrateVerb is still creating its own tracer that is used and writes to the dehydrate log.

Thanks for all of the details here!  The `{ "Output", commandOutput.ToString() }` was what was really throwing me off, but it looks like it's probably never gone where we expected (after updating `clone` to dispose its tracer).

The latest changes look good to me, thanks!